src/navigation/components/MenuItems/tree.ts

```diff
@@ -64,35 +65,68 @@ export function getItemId(item: MenuDetails_menu_items): string {
 export function getDiff(
   originalTree: TreeItem[],
   newTree: TreeItem[]
-): TreeOperation {
+): TreeOperation[] {
   const originalMap = treeToMap(originalTree, "root");
   const newMap = treeToMap(newTree, "root");
 
-  const diff: TreeOperation[] = Object.keys(newMap).map(key => {
+  const diff: TreeOperation[] = Object.keys(newMap).flatMap(key => {
     const originalNode = originalMap[key];
     const newNode = newMap[key];
 
     const patch = getPatch(originalNode, newNode);
 
     if (patch.length > 0) {
       const addedNode = patch.find(operation => operation.type === "add");
+      const removedNode = patch.find(operation => operation.type === "remove");
+
       if (!!addedNode) {
+        const changedParent = originalNode.length !== newNode.length;
+        const sortOrder = removedNode
+          ? addedNode.newPos - removedNode.oldPos
+          : addedNode.newPos;
+
+        // This exists because backend doesn't recongize the position of the new node
+        // when it's moved from child to parent and/or up
+        // We have to make an additional move so that backend can sort the new tree correctly
```

Comment:
This does not seem very intuitive, have you proposed a better API for this, or is there no better way of handling this?

Reply:
I know it doesn't seem intuitive, I was discussing the same thing with backend. This solution allows the backend to change only one element in the hierarchy per one move. The other way would be to assign the new element at the very beginning at first and then do the move, but this would cause "a lot of problems" on the backend side - instead of only one change all elements would have to be changed. Personally, I don't think this whole navigation has been properly thought through, I'm not really proud of the frontend solution but it's how it has to be to be compliant with backend. On the other hand, forcing the solution from few years ago (absolute indexes) is not very resources and memory friendly, or so I have been told.